### PR TITLE
Force reloading of APIServer on certs change

### PR DESCRIPTION
**Pull request by csrwng**
Whenever the serving cert secret is updated (as in renewal), we need
to let the Kube API server on the target cluster know so that the
new certs are applied. This fix uses a hash of all cert secrets
passed to the API server to set the value of the
"forceDeploymentReason" field on the KubeAPIServer resource on the
cluster.

### CODE REVIEW

File: pkg/controller/controlplanecerts/controlplanecerts_controller.go [hunk 1]
```diff
@@ -297,7 +297,19 @@ func (r *ReconcileControlPlaneCerts) generateControlPlaneCertsSyncSet(cd *hivev1
 		cdLog.WithError(err).Error("cannot add typemeta to syncset resources")
 		return nil, err
 	}
+	// kubeAPIServerPatch indicates sets the forceRedeploymentField on the kube API server cluster operator
```
Comment (dgoodwin) [inline pkg/controller/controlplanecerts/controlplanecerts_controller.go]: nit: "indicates sets"

File: pkg/controller/controlplanecerts/controlplanecerts_controller.go [hunk 2]
```diff
@@ -297,7 +297,19 @@ func (r *ReconcileControlPlaneCerts) generateControlPlaneCertsSyncSet(cd *hivev1
...
+	// to a hash of all the cert secrets. If the content of the certs secrets changes, then the new
+	// hash value will force the kube API server to redeploy and apply the new certs.
```
Comment (dgoodwin) [inline pkg/controller/controlplanecerts/controlplanecerts_controller.go]: The in-cluster operator watches this annotation, and if it changes will automatically redeploy, but do nothing if the value stays steady?

File: pkg/controller/controlplanecerts/controlplanecerts_controller.go [hunk 3]
```diff
@@ -375,3 +387,28 @@ func controlPlaneCertsSyncSetName(name string) string {
...
+		keys = append(keys, k)
+	}
+	sort.StringSlice(keys).Sort()
+	fmt.Fprintf(w, "%s/%s\n", secret.Namespace, secret.Name)
+	for _, k := range keys {
+		fmt.Fprintf(w, "%s: %x\n", k, secret.Data[k])
+	}
+}
+
+func secretsHash(secrets []*corev1.Secret) string {
```
Comment (dgoodwin) [inline pkg/controller/controlplanecerts/controlplanecerts_controller.go]: This func could probably use a unit test just to make sure we're getting consistent hashes, but then I'm not entirely sure if we can control the order of the secret keys in a test?

File: pkg/controller/controlplanecerts/controlplanecerts_controller.go (hunk 2 above)
Comment (csrwng) [reply pkg/controller/controlplanecerts/controlplanecerts_controller.go]: Yes

**Comment by csrwng** [comment]
Thx @dgoodwin, comments addressed

**Pull request merged**